tests/ga/extension_emulator.py

```diff
@@ -0,0 +1,315 @@
+# Copyright 2020 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Requires Python 2.6+ and Openssl 1.0+
+#
+
+import json
+import re
+import uuid
+import os
+import contextlib
+from enum import Enum
+import subprocess
+
+import azurelinuxagent.common.utils.fileutil as fileutil
+import azurelinuxagent.common.conf as conf
+from azurelinuxagent.common.future import ustr
+from tests.tools import Mock, patch
+
+from tests.protocol.mocks import HttpRequestPredicates
+
+from azurelinuxagent.ga.exthandlers import ExtHandlerInstance, HandlerManifest
+
+
+class ExtensionCommandNames(Enum):
+    INSTALL = "install"
+    UNINSTALL = "uninstall"
+    UPDATE = "update"
+    ENABLE = "enable"
+    DISABLE = "disable"
+
+class Actions(object):
+    """
+    A collection of static methods providing some basic functionality for the ExtensionEmulator
+    class' actions.
+    """
+
+    @staticmethod
+    def succeed_action(*args, **kwargs):
+        """
+        A nop action with the correct function signature for ExtensionEmulator actions.
+        """
+        return 0
+    
+    
+    @staticmethod
+    def fail_action(*args, **kwargs):
+        """
+        A simple fail action with the correct function signature for ExtensionEmulator actions.
+        """
+        raise ExtensionError("FailAction called.")
```

Comment:
My thoughts would be delayed parsing; I don't think any tests in the test suite actually use this method; instead, they use `generate_unique_fail` (below). Do you think a better solution would be to remove this all together? 

Nice catch, btw.

Reply:
I see this is being used by this test - `test_non_enabled_ext_should_not_be_disabled_at_ver_update` but yeah if its not really needed then might as well remove it.

Reply:
Oh, interesting. The "name not found" error was basically acting as a stand in for the `ExtensionError`, which is why this was working.

I think it would make sense to remove this, if you guys agree.